lib/administrate/namespace.rb

```diff
@@ -7,6 +7,8 @@ def initialize(namespace)
     def resources
       namespace_controller_paths.uniq.map do |controller|
         controller.gsub(/^#{namespace}\//, "").to_sym
```

Comment:
Adding this second validation to this line might be a better idea in my opinion:

`controller.starts_with?(namespace.to_s) && !controller.end_with?('/')`

It removes the complexity of having to run throughout the collection twice, also it won't allow any possible sub-route root to mess the generation up